### PR TITLE
Fix ArgumentError in LegacyAppeal#address

### LINKED ISSUE #13545: ArgumentError type in LegacyAppeal
## Description

```
ArgumentError

unknown keyword: type

    app/models/address.rb in initialize at line 7

      attr_reader :country, :city, :zip, :address_line_1, :address_line_2, :address_line_3, :state

      # rubocop:disable Metrics/ParameterLists
      def initialize(
        address_line_1: nil, address_line_2: nil, address_line_3: nil,
        city:, zip: nil, country: nil, state: nil
      )
```

The BGS address service is returning a hash with the key `type`. Since this hash is being passed directly into the `Address` constructor, an `ArgumentError` is being thrown. 

## Acceptance criteria
- [ ] Reproduce and fix underlying issue

## Background/context/resources

* Sentry alert: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/8752/?referrer=webhooks_plugin
* Sentry alert: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/8753/?referrer=webhooks_plugin

Related to https://github.com/department-of-veterans-affairs/caseflow/pull/13411

### PR DESCRIPTION
**Pull request by ferristseng**
Resolves #13545 

### Description

  * Move mapping of `ptcpnt_addrs_type_nm => type` to `AddressMapper`
  * Update `address` so `Address` constructor is only called with expected args
  * Add test to ensure changes to `address` method work 

### Acceptance Criteria
- [ ] Reproduce and fix underlying issue

### CODE REVIEW

File: app/mappers/address_mapper.rb [hunk 1]
```diff
@@ -11,7 +11,8 @@ def get_address_from_bgs_address(bgs_address)
       city: bgs_address[:city_nm],
       country: bgs_address[:cntry_nm],
       state: bgs_address[:postal_cd],
-      zip: bgs_address[:zip_prefix_nbr]
```
Comment (ferristseng) [inline app/mappers/address_mapper.rb]: Moving the type mapping here`type: bgs_address[:ptcpnt_addrs_type_nm]` makes the fake `BgsService` behave like the one in production. `BgsService` is returning `type` in the address hash, which it isn't doing in the test suite -- this change makes it consistent among both environments.

File: app/services/bgs_address_service.rb [hunk 2]
```diff
@@ -11,7 +11,17 @@ class BgsAddressService
   def address
     return nil unless found?
 
-    bgs_record
+    # The address from BGS includes a type field. Filter the hash keys to only include
+    # address components (for Address#new).
+    bgs_record.slice(
```
Comment (ferristseng) [inline app/services/bgs_address_service.rb]: From the tests, it seems like `bgs_record` is already expected to only include these fields. In production, I was able to confirm that this isn't the case.

File: app/mappers/address_mapper.rb (hunk 1 above)
Comment (rubaiyat22) [reply app/mappers/address_mapper.rb]: Good idea!

**Comment by rubaiyat22** [comment]
I'm trying to understand this a little more. Was this error already existing and we never saw it or is this new?

**Comment by ferristseng** [comment]
> I'm trying to understand this a little more. Was this error already existing and we never saw it or is this new?

This is a new error introduced in https://github.com/department-of-veterans-affairs/caseflow/pull/13411. In that PR, we changed the source for the veteran and appellant's address from VACOLS to BGS. It wasn't caught by tests because the mocks for our fake `BgsService` don't match exactly what BGS is returning in prod.

File: app/services/bgs_address_service.rb (hunk 2 above)
Comment (rubaiyat22) [reply app/services/bgs_address_service.rb]: I wonder why why we're excluding the `type` field 🤔

**Pull request merged**